web/src/portico/signup.ts

```diff
@@ -342,3 +343,71 @@ $(() => {
         }
     });
 });
+
+function get_or_set(input_selector: string): (newval?: string) => string {
+    return function () {
+        const $elem = $<HTMLInputElement | HTMLTextAreaElement>(input_selector);
+
+        if (!$elem.length) {
+            return "";
+        }
+
+        const oldval = $elem.val();
+        if (oldval === undefined) {
+            return "";
+        }
+        return oldval.trim();
+    };
+}
+
+const password_input = get_or_set("input#id_password");
+const new_password1_input = get_or_set("input#id_new_password1");
+
+const $password_elem = $("input#id_password");
+const $new_password1_elem = $("input#id_new_password1");
+
+if ($password_elem.length) {
+    $password_elem.on("input", () => {
+        check_overflow_password(password_input(), true);
+    });
+}
+
+if ($new_password1_elem.length) {
+    $new_password1_elem.on("input", () => {
+        check_overflow_password(new_password1_input(), false);
+    });
+}
+
+export function check_overflow_password(input_field: string, is_signup: boolean): number {
```

Comment:
I tried and I see no problem in directly passing `$password_elem` to this function and then get the value using `val()`.

Also, I guess we can skip `is_signup` variable and directly use the input jquery element to find the indicator and button element using `.find()` and `.closest()` functions of jquery.

Reply:
> I tried and I see no problem in directly passing `$password_elem` to this function and then get the value using `val()`.
> 

Thanks for this, it worked.

> Also, I guess we can skip `is_signup` variable and directly use the input jquery element to find the indicator and button element using `.find()` and `.closest()` functions of jquery.

I tried using this, but the code becomes messier compared to the current implementation since the DOM structure of the `register` and `reset-password` pages differs.

Reply:
I guess you can just add a common class to button like `submit-button` and then do something like `closest("form").find(".submit-button")` since a form should have only one button in most of the cases.